CGAL 5.0 support

Your package fails to build with CGAL 5.0 Beta 2. I modified the top-level CMakeLists.txt as follows:
- add set(CMAKE_CXX_STANDARD 14)
- remove block with "c++98"
- remove "4.3" from find_package() call

When trying to link the unit test, it fails with
```/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/Force3DTest.cpp.o: in function `SFCGAL_algorithm_Force3DTest::testIgnoreEmpty::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/Force3DTest.cpp:57: undefined reference to `SFCGAL::algorithm::force3D(SFCGAL::Geometry&, 
CGAL::Lazy_exact_nt<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, (boost::multiprecision::expression_template_option)1> > const&)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/Force3DTest.cpp.o: in function `SFCGAL_algorithm_Force3DTest::testPointForceZ::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/Force3DTest.cpp:65: undefined reference to `SFCGAL::algorithm::force3D(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, (boost::multiprecision::expression_template_option)1> > const&)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/Force3DTest.cpp.o: in function `SFCGAL_algorithm_Force3DTest::testPointForceZWithValue::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/Force3DTest.cpp:71: undefined reference to `SFCGAL::algorithm::force3D(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, (boost::multiprecision::expression_template_option)1> > const&)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/Force3DTest.cpp.o: in function `SFCGAL_algorithm_Force3DTest::test_MixedLineString2D3D::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/Force3DTest.cpp:82: undefined reference to `SFCGAL::algorithm::force3D(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, (boost::multiprecision::expression_template_option)1> > const&)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/UnionTest.cpp.o: in function `SFCGAL_algorithm_UnionTest::PolygonPolygon2::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/UnionTest.cpp:230: undefined reference to `SFCGAL::algorithm::translate(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>)'
/usr/bin/ld: ./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/UnionTest.cpp:234: undefined reference to `SFCGAL::algorithm::translate(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/UnionTest.cpp.o: in function `SFCGAL_algorithm_UnionTest::PolygonPolygon3::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/UnionTest.cpp:259: undefined reference to `SFCGAL::algorithm::translate(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>)'
/usr/bin/ld: ./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/UnionTest.cpp:261: undefined reference to `SFCGAL::algorithm::translate(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/UnionTest.cpp.o: in function `SFCGAL_algorithm_UnionTest::VolumeVolume::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/UnionTest.cpp:363: undefined reference to `SFCGAL::algorithm::translate(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>)'
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/algorithm/UnionTest.cpp.o:./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/algorithm/UnionTest.cpp:370: more undefined references to `SFCGAL::algorithm::translate(SFCGAL::Geometry&, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>, CGAL::Lazy_exact_nt<CGAL::Gmpq>)' follow
/usr/bin/ld: CMakeFiles/unit-test-SFCGAL.dir/SFCGAL/transform/ForceZOrderPointsTest.cpp.o: in function `SFCGAL_transform_ForceZOrderPointsTest::simple::test_method()':
./obj-x86_64-linux-gnu/test/unit/./test/unit/SFCGAL/transform/ForceZOrderPointsTest.cpp:40: undefined reference to `SFCGAL::transform::ForceZOrderPoints::ForceZOrderPoints(CGAL::Lazy_exact_nt<boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, (boost::multiprecision::expression_template_option)1> > const&)'
collect2: error: ld returned 1 exit status```

The following patch fixed the PostGIS build failure:
```
Description: Add gmpxx to `sfcgal-config --libs`.
 Fixes link errors:
 .
 /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libSFCGAL.so: undefined reference to `operator>>(std::istream&, __mpz_struct*)'
 /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libSFCGAL.so: undefined reference to `operator<<(std::ostream&, __mpz_struct const*)'
Author: Bas Couwenberg <sebastic@debian.org>

--- a/sfcgal-config.in
+++ b/sfcgal-config.in
@@ -38,7 +38,7 @@ case $1 in
        echo -I${prefix}/include 
        ;;
     --libs)
-       echo -L${libdir} -l@SFCGAL_LIB_NAME@
+       echo -L${libdir} -l@SFCGAL_LIB_NAME@ -lgmpxx
        ;;
     --ldflags)
        echo -L${libdir}
```
The is similar to the link issue with the tests for which @joachim-reichel provided this patch:

https://salsa.debian.org/debian-gis-team/sfcgal/blob/debian/1.3.7-3/debian/patches/fix-linker-error.patch

--- a/.travis.yml
+++ b/.travis.yml
@@ -11,7 +11,7 @@ compiler:
   - clang
 
 env:
-  - CGAL_VERSION=4.13
+  - CGAL_VERSION=5.0.2
 
 before_install:
   - ./travis/${TRAVIS_OS_NAME}/before_install.sh $CGAL_VERSION

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required( VERSION 2.8 )
 project( SFCGAL )
 
 set( CMAKE_DEBUG_POSTFIX "d" )
+set(CMAKE_CXX_STANDARD 14)
 
 #----------------------------------------------------------------------------
 # build options
@@ -53,6 +54,10 @@ endif()
 # 4.13 recommended
 find_package( CGAL COMPONENTS Core REQUIRED )
 message( STATUS "CGAL ${CGAL_VERSION} found" )
+
+if( "${CGAL_VERSION}" VERSION_GREATER_EQUAL "5.0.0")
+add_definitions( "-DCGAL_USE_GMPXX=1" )
+endif()
 
 include_directories( ${CMAKE_BINARY_DIR}/include )
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -50,6 +50,10 @@ set_target_properties( SFCGAL PROPERTIES VERSION ${SFCGAL_VERSION}
 
 target_link_libraries( SFCGAL CGAL::CGAL CGAL::CGAL_Core)
 
+if( "${CGAL_VERSION}" VERSION_GREATER_EQUAL "5.0.0")
+target_link_libraries( SFCGAL gmpxx )
+endif()
+
 target_link_libraries( SFCGAL ${Boost_LIBRARIES} )
 
 if ( ${Use_precompiled_headers} )

--- a/src/config.h.cmake
+++ b/src/config.h.cmake
@@ -21,7 +21,6 @@
 #define _SFCGAL_CONFIG_H_
 
 #define CGAL_DO_NOT_USE_BOOST_MP 1
-#define CGAL_DO_NOT_USE_GMPXX 1
 
 #include <SFCGAL/export.h>
 

